docs/usage/Context.md

```diff
@@ -0,0 +1,55 @@
+# Context
+
+The context is a configuration resource containing shared configuration for installations.
+This config can contain the repository context, registry pull secrets, or even deployer specific context.
+
+A context can be referenced by installations in the same namespace.
+
+## Basic structure
+
+A context object contains the repository context and an optional list of registry pull secrets.
+These registry pull secrets are references to secrets in the same namespace as the context that contains oci registry access credentials.
+These credentials can be used to access component descriptors, blueprints or even deployable artifacts like helm charts or oci images.
+
+```yaml
+apiVersion: landscaper.gardener.cloud/v1alpha1
+kind: Context
+metadata:
+  name: default
+  namespace: default
+
+repositoryContext:
+  type: ociRegistry
+  baseUrl: "example.com"
+
+registryPullSecrets: # additional pull secrets to access component descriptors and blueprints
+- name: my-pullsecret
+```
+
+## Default Context
+
+Just like kubernetes creates a default service account in every namespace, a default context is created in every namespace.
+The default context can be configured in the landscaper configuration `.controllers.context.config.default` as described in the [example](../../examples/00-Landscaper-Configuration.yaml).
+
+The default context object can also be manually modified as the landscaper controller only reconciles configured defaults.
+
+## Configuration
+
+The context controller can be configured in the landscaper config `.controllers.context`.
+
+```yaml
+apiVersion: config.landscaper.gardener.cloud/v1alpha1
+kind: LandscaperConfiguration
+
+controllers:
+  context:
+    config:
+      default:
+        disable: false
+        excludeNamespaces: # optional
+        - kube-system
+        - ls-system
+        repositoryContext: # define the default repository context for installations
+          type: ociRegistry
+          baseUrl: "myregistry.com/components"
+```
```

Comment:
I got some questions one my mind while reading this.
The landscaper will create a default context in each namespace except it is excluded in `excludeNamespaces`.
What about non-default contexts? I can specify them in the landscaper configuration, right? Are the also created by the landscaper or do they have to be created manually? In which namespace are they living or do they have  to be created manually by the user?

Reply:
non-defaults contexts can be manually created by the users using e.g. `kubectl apply`.
There are alos just a resource like installations. so they are namespaced CR's and teh namespaced is spcified during creation

Reply:
But why would we then add them to the landscaper configuration?
My impression was, I could specify them like this:

```yaml
controllers:
  context:
    config:
     myContext:
       repositoryContext:
         type: "ociRegistry"
         baseUrl: ""

registryConfig:
  secrets:
    myContext: {}
```

Or am I wrong?

Reply:
no you can only configure the default context that is then populated to all namespaces.
Other contexts should be added manually.
It comparable to the service accounts and the default service account in core k8s.

Reply:
I think it will be clearer as soon as this PR is also added: https://github.com/gardener/landscaper/pull/335

Then the structure would look like:
```
controllers:
  contexts:
    workers: 5
    config:
      default:
        repositoryContext: ...
```

But we could also rename the `default` key to `defaultContext`. Maybe that will also make it clearer.
WDYT?

Reply:
Ah, ok. No, then i think it is OK to leave it like it is.
I think we just need some good tutorials on how to handle/manage contexts.